tla-io/src/main/scala/at/forsyte/apalache/io/annotations/parser/AnnotationLexer.scala

```diff
@@ -61,37 +57,21 @@ object AnnotationLexer extends RegexParsers {
 
   private def string: Parser[STRING] = {
     """"[a-zA-Z0-9_~!@#\\$%^&*\-+=|(){}\[\],:;`'<>.?/ \t\r\f\n]*"""".r ^^ { name =>
-      STRING(removeLeadingCommentsAndControlChars(name.substring(1, name.length - 1)))
+      STRING(name.substring(1, name.length - 1))
     }
   }
 
   private def inline_string: Parser[INLINE_STRING] = {
-    """:[a-zA-Z0-9_~!#\\$%^&*\-+=|(){}\[\],:`'<>.?/ \t\r\f\n]*;""".r ^^ { name =>
-      INLINE_STRING(removeLeadingCommentsAndControlChars(name.substring(1, name.length - 1)))
+    """:[a-zA-Z0-9_~!#\\$%^&*\-+=|(){}\[\],:`'<>.?/ \t\r\f\n]*;""".r ^^ { text =>
+      val contents = text.substring(1, text.length - 1)
+      // Inline string may contain line feeds and other control characters. Remove them.
+      val cleared = whiteSpace.replaceAllIn(contents, " ")
+      INLINE_STRING(cleared)
     }
   }
 
-  /**
-   * <p>It is not uncommon to write multiline strings in annotations. Since these strings are written in comments,
-   * it should be possible to write them like:</p>
-   *
-   * <pre>
-   *
-   * \* @type: Int
-   * \*          => Int;
-   * \*
-   * \* @type("Int
-   * \*          => Int")
-   * </pre>
-   *
-   * <p>This method removes the leading "\*" from a multiline string. Additionally, it replaces with a space
-   * a group of carriage returns, form feeds, line feeds, and tabs.</p>
-   *
-   * @param text text to preprocess
-   * @return text without leading "\*" and ASCII control characters
-   */
-  private def removeLeadingCommentsAndControlChars(text: String): String = {
-    text.replaceAll("""\n[ \t\r\f]*\\\*""", "").replaceAll("[\n\t\r\f]*", "")
+  private def unexpected_char: Parser[Nothing] = {
+    failure("Unexpected character. Missing ')' or ';'?")
   }
```

Comment:
Could we return what the unexpected character is? I think that would help identify the source of the problem.

Reply:
I tried to figure out how to do it with Scala parser combinators, and I failed. I can produce `failure` or `err`, but neither of them accepts a token.

Reply:
This seems to work:

```suggestion
  private def unexpected_char: Parser[Nothing] = {
    (".".r ^^ { char =>
      s"Unexpected character '${char}'. Missing ')' or ';'?"
    }).flatMap(failure)
  }
```

Reply:
Oh, that's a good one. Thanks a lot!